terraform/github/import_resources.py

```
@@ -113,111 +112,136 @@ def __str__(self) -> str:
     complete_resource_address = f"{resource_address}[\"{index_key}\"]" if index_key else resource_address
     cmd = ["terraform", "import", complete_resource_address, resource_id]
+    print("\trun =>", *cmd)
     if not is_dry_run:
-        output = subprocess.run(cmd, capture_output=True)
-        print(output.stdout.decode())
-    else:
-        print(f"import_missing_resource(resource_address: {resource_address}, resource_id: {resource_id}, index_key: {index_key}",
-              "=>\n\t\033[1m\033[31m", *cmd, "\033[0;0m", end="\n\n")
+        output = subprocess.run(cmd, capture_output=True, check=False)
+        try:
+            output.check_returncode()
+        except subprocess.CalledProcessError as error:
+            if "Could not find" in error.stderr.decode():
+                print(error.stderr.decode().splitlines()[1])
+            else:
+                print(error.stderr.decode())
+        else:
+            if "Import successful!" in output.stdout.decode():
+                print("Import Successful!")
+            else:
+                print(error.stderr.decode())
+    print("\n", end="")
 
 
 def get_default_branches() -> dict[str, str]:
     branches = {}
     cmd = ["terraform", "state", "pull"]
-    complete_path = Path(__file__).parent.joinpath("terraform.state.pull")
-    tfstate_file = open(complete_path, "w", encoding="utf-8")
-    subprocess.run(cmd, stdout=tfstate_file, encoding="utf-8")
-    tfstate_json = json.load(open(complete_path, "r", encoding="utf-8"))
-    for repository in tfstate_json["resources"][0]["instances"]:
-        branches[repository["index_key"]] = repository["attributes"]["default_branch"]
+    print("Fetching latest remote state file =>", *cmd)
+    output = subprocess.run(cmd, capture_output=True, check=False)
+    try:
+        output.check_returncode()
+    except subprocess.CalledProcessError:
+        print(output.stderr.decode())
```

This error will be ignored

---

There is the possibility of `terraform state pull` failing due to network issue as it has to pull this from Terraform Cloud.